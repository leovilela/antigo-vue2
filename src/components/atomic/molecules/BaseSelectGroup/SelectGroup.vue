<template>
  <div>
    <label class="select" for="" v-if="label">{{label}}</label>
    <Base-Select :selectgroup="selectgroup" @input="$emit('input', $event.target.value)" />
  </div>
</template>

<script>
export default {
  props: {
    selectgroup: {
      type: [String, Object, Array],
      required: true,
    },
    label: {
      type: String,
      default: null,
    },
  },
};
</script>

<style scoped>
label.select {
  display: block;

}

select {
  min-width: auto;
  padding: 0;

}
</style>
