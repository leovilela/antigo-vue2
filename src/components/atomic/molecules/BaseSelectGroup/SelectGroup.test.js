import { mount } from "@vue/test-utils";
import SelectGroup from "../BaseSelectGroup/SelectGroup";

const factory = () => mount(SelectGroup, {
  propsData: {
    selectgroup: {
      id: 1,
      options: [],
    },
  },
  stubs: ["Base-Select"],
});

describe("SelectGroup", () => {
  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
