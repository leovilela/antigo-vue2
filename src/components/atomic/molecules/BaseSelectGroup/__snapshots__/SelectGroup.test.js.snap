// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectGroup renders properly 1`] = `
<div>
  <!---->
  <base-select-stub selectgroup="[object Object]"></base-select-stub>
</div>
`;
